aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -110,11 +110,7 @@
 
   indexFunction += getBitSizeString(scalar_type);
   if (accumulate) {
-    TORCH_CHECK(scalar_type == ScalarType::Float || scalar_type == ScalarType::Int,
-                "Unsupported data type for accumulate case: ",
-                getMPSTypeString(scalar_type));
-    string dtypeString = (scalar_type == ScalarType::Float) ? "_float" : "_int";
-    indexFunction += dtypeString;
+    indexFunction += "_" + scalarToMetalTypeString(scalar_type);
```

Comment:
Nit split into append char and then append string for mildly faster code (and avoiding single char strings).

Reply:
Thank you for the feedback, I plan to refactor this PR a bit more to include unit tests for atomic ops